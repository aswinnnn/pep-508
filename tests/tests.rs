use insta::assert_debug_snapshot;

use pep_508::parse;

#[test]
fn basic() {
    assert_debug_snapshot!(parse("requests"));
    assert_debug_snapshot!(parse("beautifulsoup4"));
    assert_debug_snapshot!(parse("Click>=7.0"));
    assert_debug_snapshot!(parse("tomli;python_version<'3.11'"));

    assert!(parse("").is_err());
    assert!(parse("# comment").is_err())
}

// tests from the PEP
#[test]
fn pep() {
    assert_debug_snapshot!(parse("A"));
    assert_debug_snapshot!(parse("A.B-C_D"));
    assert_debug_snapshot!(parse("aa"));
    assert_debug_snapshot!(parse("name"));
    assert_debug_snapshot!(parse("name<=1"));
    assert_debug_snapshot!(parse("name>=4"));
    assert_debug_snapshot!(parse("name>=3,<2"));
    assert_debug_snapshot!(parse("name@http://foo.com"));
    assert_debug_snapshot!(parse(
        "name [fred,bar] @ http://foo.com ; python_version=='3.7'"
    ));
    assert_debug_snapshot!(parse(
        "name[quux, strange];python_version<'3.7' and platform_version=='2'"
    ));
    assert_debug_snapshot!(parse("name; os_name=='a' or os_name=='b'"));
    assert_debug_snapshot!(parse("name; os_name=='a' and os_name=='b' or os_name=='c'"));
    assert_debug_snapshot!(parse(
        "name; os_name=='a' and (os_name=='b' or os_name=='c')"
    ));
    assert_debug_snapshot!(parse("name; os_name=='a' or os_name=='b' and os_name=='c'"));
    assert_debug_snapshot!(parse(
        "name; (os_name=='a' or os_name=='b') and os_name=='c'"
    ));
}
