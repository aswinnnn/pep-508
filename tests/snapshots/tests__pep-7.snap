---
source: tests/tests.rs
expression: "parse(\"name>=3,<2\")"
---
Ok(
    Dependency {
        name: "name",
        extras: [],
        spec: Some(
            Version(
                [
                    VersionSpec {
                        comparator: Ge,
                        version: "3",
                    },
                    VersionSpec {
                        comparator: Lt,
                        version: "2",
                    },
                ],
            ),
        ),
        marker: None,
    },
)
